03 Writing Algorithms/03 Securities/99 Asset Classes/02 Equity Options/02 Handling Data/99 Examples.html

```diff
@@ -0,0 +1,200 @@
+        self.set_security_initializer(lambda security: FuncSecuritySeeder(self.get_last_known_prices).seed_security(security))
+        # Subscribe to underlying data for ATM calculation using the update underlying price.
+        # Set data normalization mode to raw is required to ensure strike price and underlying price is comparable.
+        spy = self.add_equity("SPY", data_normalization_mode=DataNormalizationMode.RAW).symbol
+        # Subscribe to SPY option data.
+        option = self.add_option(spy)
+        self._symbol = option.symbol
+        # We wish to only trade the contracts expiring within the same week since they have the highest volume.
+        option.set_filter(lambda u: u.include_weeklys().contracts(lambda x: [s for s in x if s.id.date &lt;= Expiry.end_of_week(self.time)]))
+    
+    def on_data(self, slice: Slice) -&gt; None:
+        # Only want to obtain the option chain of the selected symbol.
+        chain = slice.option_chains.get(self._symbol)
+        if not chain:
+            return
+    
+        for contract in chain:
+            # Mid price = average of bid close price and ask close price
+            mid_price = (contract.bid_price + contract.ask_price) * 0.5</pre>
+</div>
+
+<h4>Example 3: Get Instant Delta</h4>
+<p>The option greeks change rapidly, so we need to obtain the instant greeks to accurately calculate the hedge size for arbitration or replication portfolio. You can call the <code>Greeks</code> property from the <code>OptionChain</code> data to access various greeks. In this example, we will demonstrate how to obtain the contract with delta closest to 0.4 among all call contracts expiring the same week.</p>
+<div class="section-example-container">
+	<pre class="csharp">public class EquityOptionHandlingDataAlgorithm : QCAlgorithm
+{
+    private Symbol _symbol;
+        
+    public override void Initialize()
+    {
+        // Seed the price with last known price to ensure the underlying price data is available on initial option contract filtering.
+        SetSecurityInitializer((security) =&gt; new FuncSecuritySeeder(GetLastKnownPrices).SeedSecurity(security));
+        // Subscribe to underlying data for ATM calculation using the update underlying price.
+        // Set data normalization mode to raw is required to ensure strike price and underlying price is comparable.
+        var spy = AddEquity("SPY", dataNormalizationMode: DataNormalizationMode.Raw).Symbol;
+        // Subscribe to SPY option data.
+        var option = AddOption(spy);
+        _symbol = option.Symbol;
+        // We wish to only trade the contracts expiring within the same week since they have the highest volume.
+        // 0.4 Delta will only appear in call contracts.
+        option.SetFilter((u) =&gt; u.IncludeWeeklys().CallsOnly().Contracts((x) =&gt; x.Where(s =&gt; s.ID.Date &lt;= Expiry.EndOfWeek(Time))));
```

Comment:
We could use the Delta method:
```csharp
option.SetFilter((u) =&gt; u.IncludeWeeklys().Delta(0.2m, 0.6m).Contracts((x) =&gt; x.Where(s =&gt; s.ID.Date &lt;= Expiry.EndOfWeek(Time))));
```
A range of +/-0.2 should always include 0.4 when the market opens, but we can increase to `Delta(0.1m, 0.7m)`.